src/index.js

```diff
@@ -0,0 +1,336 @@
+					imgSelectOptions: simple_local_avatar_calculate_image_select_options,
+					control: cropControl,
+				}),
+			],
+		});
+
+		/**
+		 * After the image has been cropped, apply the cropped image data to the setting
+		 *
+		 * @param {object} croppedImage Cropped attachment data.
+		 */
+
+		simple_local_avatar_frame.on('cropped', function (croppedImage) {
+			const { url } = croppedImage;
+			const attachmentId = croppedImage.id;
+			const w = croppedImage.width;
+			const h = croppedImage.height;
+
+			simple_local_avatar_set_image_from_url(url, attachmentId, w, h);
+		});
+
+		/**
+		 * If cropping was skipped, apply the image data directly to the setting.
+		 */
+		simple_local_avatar_frame.on('skippedcrop', function (selection) {
+			const url = selection.get('url');
+			const w = selection.get('width');
+			const h = selection.get('height');
+
+			simple_local_avatar_set_image_from_url(url, selection.id, w, h);
+		});
+
+		/**
+		 * After an image is selected in the media modal, switch to the cropper
+		 * state if the image isn't the right size.
+		 */
+		simple_local_avatar_frame.on('select', function () {
+			const avatarAttachment = simple_local_avatar_frame
+				.state()
+				.get('selection')
+				.first()
+				.toJSON();
+
+			if (
+				cropControl.params.width === avatarAttachment.width &&
+				cropControl.params.height === avatarAttachment.height &&
+				!cropControl.params.flex_width &&
+				!cropControl.params.flex_height
+			) {
+				avatarAttachment.dst_width = avatarAttachment.width;
+				avatarAttachment.dst_height = avatarAttachment.height;
+				simple_local_avatar_set_image_from_attachment(avatarAttachment);
+				simple_local_avatar_frame.close();
+			} else {
+				simple_local_avatar_frame.setState('cropper');
+			}
+		});
+
+		simple_local_avatar_frame.open();
+	});
+
+	/**
+	 * If the Local Avatar is removed and set to the  default one
+	 */
+	$('#simple-local-avatar-remove').on('click', function (event) {
+		event.preventDefault();
+
+		if (avatar_working) return;
+
+		avatar_lock('lock');
+		$.get(ajaxurl, {
+			action: 'remove_simple_local_avatar',
+			user_id: i10n_SimpleLocalAvatars.user_id,
+			_wpnonce: i10n_SimpleLocalAvatars.deleteNonce,
+		})
+			.done(function (data) {
+				if (data != '') {
+					avatar_container.innerHTML = data;
+					$('#simple-local-avatar-remove').hide();
+					avatar_ratings.disabled = true;
+				}
+			})
+			.always(function () {
+				avatar_lock('unlock');
+			});
+	});
+
+	/**
+	 * Update the Local Avatar image
+	 */
+
+	avatar_input.on('change', function (event) {
+		avatar_preview.attr('srcset', '');
+		avatar_preview.attr('height', 'auto');
+		URL.revokeObjectURL(avatar_blob);
+		if (event.target.files.length > 0) {
+			avatar_blob = URL.createObjectURL(event.target.files[0]);
+			avatar_preview.attr('src', avatar_blob);
+		} else {
+			avatar_preview.attr('src', current_avatar);
+		}
+	});
+});
+
+function avatar_lock(lock_or_unlock) {
```

Comment:
Can you please add short documentation above the function as you've already done very well for others?